### PR TITLE
EOS-27987: Add entrypoint script for RGW container

### PR DESCRIPTION
**Pull request by selvakumaar5496**
Signed-off-by: Selvakumar <selva.k.nambi@seagate.com>

# Problem Statement
- Add entrypoint script for RGW container

# Design
-  Removed mocking code from provisioner's test framework and updated the entrypoint for cortx-rgw container

# Coding
   Checklist for Author
-  [x] Coding conventions are followed and code is consistent

# Testing 
  Checklist for Author
- [ ] Unit and System Tests are added
- [ ] Test Cases cover Happy Path, Non-Happy Path and Scalability
- [x] Testing was performed with RPM

# Impact Analysis
  Checklist for Author/Reviewer/GateKeeper
- [ ] Interface change (if any) are documented
- [ ] Side effects on other features (deployment/upgrade)
- [ ] Dependencies on other component(s)

# Review Checklist 
  Checklist for Author
- [x] JIRA number/GitHub Issue added to PR
- [x] PR is self reviewed
- [x] Jira and state/status is updated and JIRA is updated with PR link
- [x] Check if the description is clear and explained

# Documentation
  Checklist for Author
- [ ] Changes done to WIKI / Confluence page / Quick Start Guide

### CODE REVIEW

File: src/setup/rgw.py [hunk 1]
```diff
@@ -178,6 +178,20 @@ def init(conf: MappedConf):
             Conf.delete(rgw_consul_idx, rgw_lock_key)
             Log.info(f'{rgw_lock_key} key is deleted')
 
+        Log.info('Starting radosgw service.')
```
Comment (sachitanands) [inline src/setup/rgw.py:181]: Could you please create a separate class for entrypoint logic ? 
in future if we want to add more logic to entrypoint then we should be easily isolate entry point code.

Comment (cdeshmukh) [reply src/setup/rgw.py:181]: Agree, separate function for user creation is also needed

File: src/setup/rgw.py [hunk 2]
```diff
@@ -178,6 +178,20 @@ def init(conf: MappedConf):
...
+        try:
+            os.system("sh /opt/seagate/cortx/rgw/bin/rgw_service")
```
Comment (cdeshmukh) [inline src/setup/rgw.py:183]: As discussed, keep a TODO to replace os.system with SimpleProcess calls. i know yday we tried and ran into issues, but we should not miss this, as it has security concerns

File: src/setup/rgw_service [hunk 3]
```diff
@@ -0,0 +1,22 @@
...
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+# GNU Affero General Public License for more details.
+# You should have received a copy of the GNU Affero General Public License
+# along with this program. If not, see <https://www.gnu.org/licenses/>.
+# For any questions about this software or licensing,
+# please email opensource@seagate.com or cortx-questions@seagate.com.
+
+function start_rgw {
+  machine_id=$(cat /etc/machine-id)
+  exec sudo /usr/bin/radosgw -f --name client.rgw -c /etc/cortx/rgw/"$machine_id"/cortx_rgw.conf --no-mon-config &> /var/log/ceph/rgw.log &
```
Comment (cdeshmukh) [inline src/setup/rgw_service:19]: keep TODO to use log dir from config

File: src/setup/rgw.py (hunk 2 above)
Comment (selvakumaar5496) [reply src/setup/rgw.py]: Ack

File: src/setup/rgw_service (hunk 3 above)
Comment (selvakumaar5496) [reply src/setup/rgw_service:20]: Ack

File: src/setup/rgw.py (hunk 1 above)
Comment (selvakumaar5496) [reply src/setup/rgw.py]: Separate function for user creation already exists. Separate class for entry point is added

**Review by cdeshmukh — APPROVED** [review]
Am little confused on number of files in setup dir, without filenames or description much idea.
Also src/setup/rgw_service is the bash script, which we may do away in future
This PR is mainly to bring main branch inline with code we used to showcase rgw deployment, so other teams can use fast. And in future PRs lets optimize.

**Review by sachitanands — APPROVED** [review]
looks good

**Pull request merged**